dolly-backend-app/src/main/java/no/nav/dolly/mapper/strategy/ArenaMappingStrategy.java

```diff
@@ -30,10 +44,104 @@ public void mapAtoB(Arenadata arenadata, ArenaNyBruker arenaNyBruker, MappingCon
+                                            arenadata.getDagpenger().stream()
+                                                    .filter(Objects::nonNull)
+                                                    .map(RsArenaDagpenger::getFraDato))
+                                            .flatMap(Stream::distinct)
+                                            .map(LocalDateTime::toLocalDate)
+                                            .min(LocalDate::compareTo)
+                                            .orElse(null));
+                        }
+                        if (arenadata.getAap().isEmpty()) {
+                            arenaNyBruker.setAap(null);
+                        }
+                        if (arenadata.getAap115().isEmpty()) {
+                            arenaNyBruker.setAap115(null);
+                        }
+                    }
+                })
+                .byDefault()
+                .register();
+
+        factory.classMap(Arenadata.class, ArenaDagpenger.class)
+                .customize(new CustomMapper<>() {
+                    @Override
+                    public void mapAtoB(Arenadata arenadata, ArenaDagpenger arenaDagpenger, MappingContext context) {
+                        RsArenaDagpenger rsArenaDagpenger = arenadata.getDagpenger().get(0);
+
+                        NyeDagp dagpenger = new NyeDagp();
+
+                        dagpenger.setVedtaksperiode(ArenaDagpenger.Vedtaksperiode.builder()
+                                .fom(rsArenaDagpenger.getFraDato().toLocalDate())
+                                .build());
+
+                        dagpenger.setRettighetKode(rsArenaDagpenger.getRettighetKode());
+
+                        dagpenger.setDagpengeperiode(ArenaDagpenger.Dagpengeperiode.builder()
+                                .nullstillPeriodeteller("J")
+                                .nullstillPermitteringsteller("N")
+                                .nullstillPermitteringstellerFisk("N")
+                                .build());
+
+                        dagpenger.setGodkjenningerReellArbeidssoker(ArenaDagpenger.GodkjenningerReellArbeidssoker.builder()
+                                .godkjentDeltidssoker("N")
+                                .godkjentLokalArbeidssoker("N")
+                                .godkjentUtdanning("J")
+                                .build());
+
+                        dagpenger.setTaptArbeidstid(ArenaDagpenger.TaptArbeidstid.builder()
+                                .anvedtRegelKode("GJSNITT12MND")
+                                .fastsattArbeidstid(30)
+                                .naavaerendeArbeidstid(0)
+                                .build());
+
+                        dagpenger.setUtfall("JA");
+
+                        dagpenger.setVedtaktype("O");
+
+                        dagpenger.setVilkaar(List.of(
+                                new ArenaDagpenger.Vilkaar("GEOMOB", "J"),
```

Comment:
Kan denne liste bygges en gang til en static variabel og bare hentes her, istedet for å bygge denne liste hver gang?

Reply:
Ja lurt, skal få fikset 👍